src/main/java/duke/misc/Storage.java

```diff
@@ -0,0 +1,217 @@
+ */
+public class Storage {
+    public static final String DATA_FOLDER_PATH = "./data";
+    public static final String DATA_PATH = "./data/data.txt";
+
+    /**
+     * Initialises a new storage instance.
+     */
+    public Storage() {
+        try {
+            initFileFolder();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Creates new data file/folder if it does not exists.
+     *
+     * @throws IOException Failed to create data file/folder in the correct location.
+     */
+    public static void initFileFolder() throws IOException {
+        // Check if folder exist
+        File dataFolder = new File(DATA_FOLDER_PATH);
+        // Create new folder if it does not exist
+        if (!dataFolder.isDirectory()) {
+            boolean isSuccess = dataFolder.mkdirs();
+            if (!isSuccess) {
+                throw new IOException("     Failed to create folder.\n"
+                        + "     Please create a folder named 'data' in src manually"
+                        + "     , before running this program!");
+            }
+        }
+
+        // Check if file exists
+        File dataFile = new File(DATA_PATH);
+        // Create new data.txt file if it does not exist
+        if (!dataFile.isFile()) {
+            boolean isSuccess = dataFile.createNewFile();
+            if (!isSuccess) {
+                throw new IOException("     Failed to create datafile.\n"
+                        + "     Please create a data.txt in src/data manually"
+                        + "     , before running this program!");
+            }
+        }
+    }
+
+    /**
+     * Initialises the task list according to the size specified.
+     * Copies content from data file if it exists on the hard-disk.
+     *
+     * @param size Size of tasklist
+     * @return A TaskList that may contain stored data from previous runs.
+     */
+    public static TaskList initTaskList(int size) {
+        File dataFile = new File(DATA_PATH);
+        TaskList taskListOfTasks = new TaskList(size);
+        if (dataFile.length() != 0) {
+            loadData(dataFile, taskListOfTasks);
+        }
+        return taskListOfTasks;
+    }
+
+    /**
+     * Prints each line in the data file.
+     *
+     * @throws FileNotFoundException If file is not found.
+     */
+    public static String readData() throws FileNotFoundException {
+        File dataFile = new File(DATA_PATH);
+        String result = "";
+        Scanner s = new Scanner(dataFile);
+        result += "     Here are the records in the hard disk:\n";
+        while (s.hasNext()) {
+            result += "     " + s.nextLine() + "\n";
+        }
+        return result;
+    }
+
+    /**
+     * Loads data from the data.txt file in the TaskList.
+     *
+     * @param f File that contains the data.
+     * @param listOfTasks Target TaskList to store the data.
+     */
+    public static void loadData(File f, TaskList listOfTasks) {

```

Comment:
Rather long method, maybe can try to abstract some functionalities like the splitting of strings?